src/server/services/generation/generation.service.ts

```diff
@@ -97,8 +97,42 @@ export const getGenerationResources = async ({
   user,
   supported,
 }: GetGenerationResourcesInput & { user?: SessionUser }): Promise<Generation.Resource[]> => {
+  const preselectedVersions: number[] = [];
+  if (!ids && !query) {
+    const featuredCollection = await dbRead.collection
+      .findFirst({
+        where: { userId: -1, name: 'Generator' },
+        select: {
+          items: {
+            select: {
+              model: {
+                select: {
+                  name: true,
+                  type: true,
+                  modelVersions: {
+                    select: { id: true, name: true },
+                    where: { status: 'Published' },
+                    orderBy: { index: 'asc' },
+                    take: 1,
+                  },
```

Comment:
Performance wise, this is a lot heavier than it seems - Prisma is bad at making relations so it'll perform at least 3-4 queries to get there. I suggest you write a raw query there or at least: 
- 1st: fetch the collection
- 2nd: On a separte query, get the items + model versions, etc. 

Otherwise, it's a waste of resources.

Reply:
For now I believe it won't hurt since the collection will be fairly light in items